components/CharmCalculator.tsx

```diff
@@ -0,0 +1,219 @@
+  const [selectedHero, setSelectedHero] = useState<String | null>(null);
+
+  const [availableCharms, setAvailableCharms] = useState<Charm[] | null>(null);
+  const [selectedLeftCharm, setSelectedLeftCharm] = useState<Charm | null>(null);
+  const [selectedMidCharm, setSelectedMidCharm] = useState<Charm | null>(null);
+  const [selectedRightCharm, setSelectedRightCharm] = useState<Charm | null>(null);
+
+  // Methods
+  /** Sets the current class and clears the currently selected hero and available charms
+   *  Sets available heroes based on selected class
+   *
+   * @param chosenClass | String
+   */
+  function setClass(chosenClass: String): void {
+    // set class, then set available heroes based on that class
+    setSelectedClass(chosenClass);
+    setSelectedHero(null);
+    setAvailableCharms(null);
+
+    Heroes.Classes.find((classType) => {
+      if (classType.Class === chosenClass) {
+        setAvailableHeroes(classType.Heroes.map((hero) => hero.Name));
+      }
+    });
+  }
+
+  /** Sets the current hero and clears the selected charms
+   *  Sets the available charms based on selected hero
+   *
+   * @param heroName | String
+   */
+  function setHero(heroName: String): void {
+    // set hero, then get list of available charms based on that hero
+    setSelectedHero(heroName);
+    setSelectedLeftCharm(null);
+    setSelectedMidCharm(null);
+    setSelectedRightCharm(null);
+
+    Heroes.Classes.find((classType) => {
+      if (classType.Class === selectedClass) {
+        classType.Heroes.find((hero) => {
+          if (hero.Name === heroName) {
+            let charms: Charm[] = [];
+            hero.Charms.forEach((charm) => {
+              charms.push(new Charm(charm.Name, charm.Stats));
+            });
+            setAvailableCharms(charms);
+          }
+        });
+      }
+    });
+  }
+
+  /** Sets the charm that you selected for the given position
+   *  Any time a charm is changed, recalculate the summary
+   * 
+   * @param charmName | String
+   * @param position | CharmPosition (enum) | Left/Mid/Right
+   */
+  function setCharm(charmName: string, position: CharmPosition): void {
+    // Get charm and set that selectedCharm to the current charm.
+    availableCharms?.find((charm) => {
+      if (charm.name === charmName) {
+        switch (position) {
+          case CharmPosition.Left:
+            setSelectedLeftCharm(charm);
+            break;
+          case CharmPosition.Mid:
+            setSelectedMidCharm(charm);
+            break;
+          case CharmPosition.Right:
+            setSelectedRightCharm(charm);
+            break;
+          default:
+            break;
+        }
+      }
+    });
+
+    statSummary(selectedLeftCharm, selectedMidCharm, selectedRightCharm);
+  }
+
+  // HTML
+  return (
+    <>
+      <div id="options" className="flex flex-row flex-wrap pt-8">
+        <div className="px-2">
+          <Select onValueChange={(e) => setClass(e)}>
```

Comment:
From a UI/UX perspective, maybe fixing the width of these so they don't dynamically change based on the value length would be good!

Reply:
Also if you could make it utilise the Hero/Faction icons that could be pretty neat 👀

Reply:
> Also if you could make it utilise the Hero/Faction icons that could be pretty neat 👀

Will try to add that. I like the artifact selector in builder. The select was the quick and dirty to throw data out there.

Reply:
> From a UI/UX perspective, maybe fixing the width of these so they don't dynamically change based on the value length would be good!

Is there a tailwind class to do this? I don't know how to do it in tailwind. I would normally have a separate class for this.